packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -0,0 +1,594 @@
+
+        inner.downcast_mut()
+    }
+
+    /// Insert a value into the context of the currently bound type `T`.
+    ///
+    /// Returns the previously stored value, downcast to `U`, if no previous value was stored, or
+    /// the downcast was not possible will return [`None`]
+    pub fn insert<U: 'static>(&mut self, value: U) -> Option<Box<U>> {
+        let inner = self
+            .parent
+            .inner
+            .insert(TypeId::of::<T>(), Box::new(value))?;
+
+        inner.downcast().ok()
+    }
+
+    /// One of the most common interactions with [`HookContext`] is a counter to reference previous
+    /// frames or the content emitted during [`text()`].
+    ///
+    /// This is a utility method, which uses the other primitive methods provided to automatically
+    /// increment a counter, if the counter wasn't initialized this method will return `0`.
+    ///
+    /// [`text()`]: Self::text
+    pub fn incr(&mut self) -> isize {
+        let counter: Option<&mut isize> = self
+            .parent
+            .inner
+            .get_mut(&TypeId::of::<T>())
+            .and_then(|any| any.downcast_mut());
+
+        match counter {
+            None => {
+                // if the counter hasn't been set yet, default to `0`
+                self.parent
+                    .inner
+                    .insert(TypeId::of::<T>(), Box::new(0isize));
+                0
+            }
+            Some(ctr) => {
+                *ctr += 1;
+
+                *ctr
+            }
+        }
+    }
+
+    /// One of the most common interactions with [`HookContext`] is a counter
+    /// to reference previous frames or the content emitted during [`text()`].
+    ///
+    /// This is a utility method, which uses the other primitive method provided to automatically
+    /// decrement a counter, if the counter wasn't initialized this method will return `-1` to stay
+    /// consistent with [`incr()`].
+    ///
+    /// [`incr()`]: Self::incr
+    /// [`text()`]: Self::text
+    pub fn decr(&mut self) -> isize {
+        let counter: Option<&mut isize> = self
+            .parent
+            .inner
+            .get_mut(&TypeId::of::<T>())
+            .and_then(|any| any.downcast_mut());
+
+        match counter {
+            None => {
+                // given that increment starts with `0` (which is therefore the implicit default
+                // value) decrementing the default value results in `-1`,
+                // which is why we output that value.
+                self.parent
+                    .inner
+                    .insert(TypeId::of::<T>(), Box::new(-1isize));
+                -1
+            }
+            Some(ctr) => {
+                *ctr -= 1;
+                *ctr
+            }
+        }
+    }
+}
+
+type UInt0 = ();
+#[cfg(feature = "hooks")]
+type UInt1 = ((), UInt0);
+#[cfg(feature = "hooks")]
+type UInt2 = ((), UInt1);
+
+/// Trait to interact and inject information on [`Debug`]
```

Comment:
Do we mean to refer to the `std::fmt::Debug` trait rather than the proc macro? :)
Other pieces of docs also has this, just flagging